hazelcast-jet-core/src/main/java/com/hazelcast/jet/accumulator/LinRegAccumulator.java

```diff
@@ -0,0 +1,150 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.accumulator;
+
+import com.hazelcast.jet.JetException;
+import com.hazelcast.nio.ObjectDataOutput;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+/**
+ * Maintains the components needed to compute the linear regression on a
+ * set of {@code (long, long)} pairs. The intermediate results are held in
+ * the form of {@link BigInteger} and the finished value is a {@code
+ * double}-valued linear coefficient.
+ */
+public final class LinRegAccumulator {
+    private static final int MAX_BIGINT_LEN = 255;
+
+    private long n;
+    private BigInteger sumX;
+    private BigInteger sumY;
+    private BigInteger sumXY;
+    private BigInteger sumX2;
+
+    /**
+     * Constructs a new accumulator with all components at zero.
+     */
+    public LinRegAccumulator() {
+        this.sumX = BigInteger.ZERO;
+        this.sumY = BigInteger.ZERO;
+        this.sumXY = BigInteger.ZERO;
+        this.sumX2 = BigInteger.ZERO;
+    }
+
+    /**
+     * Creates a new accumulator with the given components. Intended only for
+     * testing and deserialization.
+     */
+    public LinRegAccumulator(long n, BigInteger sumX, BigInteger sumY, BigInteger sumXY, BigInteger sumX2) {
+        this.n = n;
+        this.sumX = sumX;
+        this.sumY = sumY;
+        this.sumXY = sumXY;
+        this.sumX2 = sumX2;
+    }
+
+    /**
+     * Accumulates a new sample.
+     */
+    public void accumulate(long x, long y) {
+        BigInteger bigX = BigInteger.valueOf(x);
+        BigInteger bigY = BigInteger.valueOf(y);
+        sumX = sumX.add(bigX);
+        sumY = sumY.add(bigY);
+        sumXY = sumXY.add(bigX.multiply(bigY));
+        sumX2 = sumX2.add(bigX.multiply(bigX));
+    }
+
+    /**
+     * Combines this accumulator with the supplied one.
+     */
+    public LinRegAccumulator combine(LinRegAccumulator that) {
+        n += that.n;
+        sumX = sumX.add(that.sumX);
+        sumY = sumY.add(that.sumY);
+        sumXY = sumXY.add(that.sumXY);
+        sumX2 = sumX2.add(that.sumX2);
+        return this;
+    }
+
+    /**
+     * Deducts the supplied accumulator from this one.
+     */
+    public LinRegAccumulator deduct(LinRegAccumulator that) {
+        n -= that.n;
+        sumX = sumX.subtract(that.sumX);
+        sumY = sumY.subtract(that.sumY);
+        sumXY = sumXY.subtract(that.sumXY);
+        sumX2 = sumX2.subtract(that.sumX2);
+        return this;
+    }
+
+    /**
+     * Computes the linear coefficient of the linear regression of the
+     * accumulated samples.
+     */
+    public double finish() {
```

Comment:
Actually,  this is correct. We'll never have n=0, as we don't emit windows, when there were no events.
With n=1, we'll have `0 / 0`, which is `NaN`, which is correct. With `n>1` and all x-values the same, we'll have `a / 0`, which is `Inf` or `-Inf`, which is also correct to emit. Anyway, I'll add this to the test.